### PR TITLE
cc ecdsa alt

### PR DESCRIPTION
**Pull request by NirEkhauz**
## Description
This PR adds the following:
- Validation of input parameters to library/ecdh.c and library/ecdsa.c
- Support for ecc alternate implementation

## Status
**READY/IN DEVELOPMENT/HOLD**

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Changelog updated
- [ ] Backported

### CODE REVIEW

File: include/mbedtls/config.h [hunk 1]
```diff
@@ -275,6 +275,8 @@
 //#define MBEDTLS_SHA1_ALT
 //#define MBEDTLS_SHA256_ALT
 //#define MBEDTLS_SHA512_ALT
+//#define MBEDTLS_ECC_ALT
```
Comment (RonEld) [inline include/mbedtls/config.h]: It should be `ECDSA_ALT`

File: library/ecdh.c [hunk 2]
```diff
@@ -87,7 +87,9 @@ int mbedtls_ecdh_compute_shared( mbedtls_ecp_group *grp, mbedtls_mpi *z,
  */
 void mbedtls_ecdh_init( mbedtls_ecdh_context *ctx )
 {
-    memset( ctx, 0, sizeof( mbedtls_ecdh_context ) );
+    if ( ctx != NULL ) {
```
Comment (RonEld) [inline library/ecdh.c]: not related to alternative ecdsa. Should be in a different PR

File: library/ecdsa.c [hunk 3]
```diff
@@ -64,7 +64,7 @@ static int derive_mpi( const mbedtls_ecp_group *grp, mbedtls_mpi *x,
 cleanup:
     return( ret );
 }
-
+#if !defined (MBEDTLS_ECC_ALT)
```
Comment (RonEld) [inline library/ecdsa.c]: Since it is only alternative for functions, and not for the whole module, it should be an alternative function implementation, and not alternative module implementation

**Comment by Patater** [comment]
Hi @NirEkhauz,

Please verify that #1027 meets your needs, and if so, we can close this PR.

Thanks

**Comment by sbutcher-arm** [comment]
Closing, as this has been approved offline.
